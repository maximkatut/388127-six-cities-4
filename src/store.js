import {applyMiddleware, createStore} from 'redux';
import {composeWithDevTools} from "redux-devtools-extension";
import thunk from 'redux-thunk';

import {ActionCreator as DataActionCreator} from './actions/data-actions.js';
import {ActionCreator as UserActionCreator} from './actions/user-actions.js';
import {createAPI} from './api';
import {AppRoute} from './const';
import history from './history';
import rootReducer from './reducers';
import {Operation as DataOperation} from './reducers/data/data-reducer';
import {AuthorizationStatus, Operation as UserOperation} from './reducers/user/user-reducer';

const onUnauthorized = () => {
  store.dispatch(UserActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH));
  history.push(`${AppRoute.LOGIN}`);
};

const onBadRequest = (err) => {
  store.dispatch(DataActionCreator.catchError(true, err));
};

const api = createAPI(onUnauthorized, onBadRequest);

export const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);

store.dispatch(DataOperation.loadOffers());
store.dispatch(UserOperation.checkAuth());
