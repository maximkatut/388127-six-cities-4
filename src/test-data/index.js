export const offers = [
  {
    appliences: [`Wi-Fi`, `Microwave`],
    bedrooms: 3,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    guests: 2,
    host: {
      avatar: `img/avatar-angelina.jpg`,
      name: `Max`,
      super: true
    },
    id: `1`,
    imagesGallery: [
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    mainImage: `img/apartment-01.jpg`,
    offerType: `House`,
    premium: true,
    price: 200,
    rate: 5,
    title: `Nice, cozy, warm big bed house`
  },
  {
    appliences: [`Wi-Fi`, `Microwave`],
    bedrooms: 4,
    coordinates: [52.3809553943508, 4.939309666406198],
    description: [`An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`],
    guests: 3,
    host: {
      avatar: `img/avatar-max.jpg`,
      name: `Anton`,
      super: false
    },
    id: `2`,
    imagesGallery: [
      `img/apartment-01.jpg`,
      `img/apartment-03.jpg`,
      `img/room.jpg`
    ],
    mainImage: `img/apartment-02.jpg`,
    offerType: `Apartment`,
    premium: true,
    price: 300,
    rate: 4,
    title: `Nice, cozy, warm big bed apartment`
  }
];

