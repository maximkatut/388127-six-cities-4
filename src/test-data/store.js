import {cities, offers} from '.';
import configureStore from 'redux-mock-store';

const mockStore = configureStore([]);

export const store = mockStore({
  activeCityName: cities[0].name,
  offers,
  cities
});
