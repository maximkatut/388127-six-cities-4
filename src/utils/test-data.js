export const offerCards = [
  {
    id: `1`,
    imageSrc: `img/apartment-02.jpg`,
    offerType: `Room`,
    premium: false,
    price: 200,
    rate: 5,
    title: `Nice, cozy, warm big bed apartment`
  },
  {
    id: `2`,
    imageSrc: `img/apartment-03.jpg`,
    offerType: `House`,
    premium: true,
    price: 300,
    rate: 4,
    title: `Nice, cozy, warm big bed apartment`
  }
];
