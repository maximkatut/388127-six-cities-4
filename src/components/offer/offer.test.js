import React from 'react';
import {Provider} from 'react-redux';
import {Router, Route} from 'react-router-dom';
import renderer from 'react-test-renderer';

import {store} from '../../test-data/store';
import history from '../../history';

import Offer from './offer.jsx';

describe(`Offer`, () => {
  it(`Offer should render correctly`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <Provider store={store}>
            <Route component={Offer}/>
          </Provider>
        </Router>
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
