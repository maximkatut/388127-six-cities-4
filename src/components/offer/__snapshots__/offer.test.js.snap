// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer Offer should render correctly 1`] = `
<div
  className="page"
>
  <div
    style={
      Object {
        "display": "flex",
        "height": "100vh",
      }
    }
  >
    <div
      className="sk-chase"
    >
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
      <div
        className="sk-chase-dot"
      />
    </div>
  </div>
</div>
`;
